code/datums/traits/negative.dm

```diff
@@ -0,0 +1,480 @@
+
+/datum/quirk/insanity/on_process()
+	if(quirk_holder.reagents.has_reagent("mindbreaker"))
+		quirk_holder.hallucination = 0
+		return
+	if(prob(2)) //we'll all be mad soon enough
+		madness()
+
+/datum/quirk/insanity/proc/madness()
+	quirk_holder.hallucination += rand(10, 25)
+
+/datum/quirk/insanity/post_add() //I don't /think/ we'll need this but for newbies who think "roleplay as insane" = "license to kill" it's probably a good thing to have
+	if(!quirk_holder.mind || quirk_holder.mind.special_role)
+		return
+	to_chat(quirk_holder, "<span class='big bold info'>Please note that your dissociation syndrome does NOT give you the right to attack people or otherwise cause any interference to \
+	the round. You are not an antagonist, and the rules will treat you the same as other crewmembers.</span>")
+
+/datum/quirk/obstructive
+	name = "Physically Obstructive"
+	desc = "You somehow manage to always be in the way. You can't swap places with other people."
+	value = -1
+	mob_trait = TRAIT_NOMOBSWAP
+	gain_text = "<span class='danger'>You feel like you're in the way.</span>"
+	lose_text = "<span class='notice'>You feel less like you're in the way.</span>"
+
+/datum/quirk/social_anxiety
+	name = "Social Anxiety"
+	desc = "Talking to people is very difficult for you, and you often stutter or even lock up."
+	value = -1
+	gain_text = "<span class='danger'>You start worrying about what you're saying.</span>"
+	lose_text = "<span class='notice'>You feel easier about talking again.</span>" //if only it were that easy!
+	medical_record_text = "Patient is usually anxious in social encounters and prefers to avoid them."
+	var/dumb_thing = TRUE
+
+/datum/quirk/social_anxiety/on_process()
+	var/nearby_people = 0
+	for(var/mob/living/carbon/human/H in oview(3, quirk_holder))
+		if(H.client)
+			nearby_people++
+	var/mob/living/carbon/human/H = quirk_holder
+	if(prob(2 + nearby_people))
+		H.stuttering = max(3, H.stuttering)
+	else if(prob(min(3, nearby_people)) && !H.silent)
+		to_chat(H, "<span class='danger'>You retreat into yourself. You <i>really</i> don't feel up to talking.</span>")
+		H.silent = max(10, H.silent)
+	else if(prob(0.5) && dumb_thing)
+		to_chat(H, "<span class='userdanger'>You think of a dumb thing you said a long time ago and scream internally.</span>")
+		dumb_thing = FALSE //only once per life
+		if(prob(1))
+			new/obj/item/reagent_containers/food/snacks/pastatomato(get_turf(H)) //now that's what I call spaghetti code
+
+//If you want to make some kind of junkie variant, just extend this quirk.
+/datum/quirk/junkie
+	name = "Junkie"
+	desc = "You can't get enough of hard drugs."
+	value = -2
+	gain_text = "<span class='danger'>You suddenly feel the craving for drugs.</span>"
+	lose_text = "<span class='notice'>You feel like you should kick your drug habit.</span>"
+	medical_record_text = "Patient has a history of hard drugs."
+	var/drug_list = list("crank", "krokodil", "morphine", "happiness", "methamphetamine") //List of possible IDs
```

Comment:
Space drugs ain't one of them?

Reply:
I am not aiming to have this trait even in, but fair - lets add space drugs.

Reply:
What about LSD? Or is that "happiness"